### ISSUE TITLE
Feature Request: Follow Maven SNAPSHOT versioning convention

### ISSUE THREAD

**Issue by daniel-shuy** [scalatest/scalatest#1766]
I would like to request that SNAPSHOT versions follow the Maven convention of including the `SNAPSHOT` qualifier (as defined in https://maven.apache.org/pom.html#Version_Order_Specification) instead of `SNAP`, eg. `3.2.0-SNAPSHOT10` instead of `3.2.0-SNAP10`.

This will allow tools like [WhiteSource Renovate](https://renovate.whitesourcesoftware.com/) to properly ignore unstable releases.

**Comment by timw** [comment]
We're running into problems with this as well, using Dendabot to detect new versions and automatically create PRs for updates.

tl;dr (and abridged version) of Maven versioning rules:
Maven knows about ordering a [set of string qualifiers](https://github.com/apache/maven/blob/master/maven-artifact/src/main/java/org/apache/maven/artifact/versioning/ComparableVersion.java#L356) that can be added to the digits of a dot separated version string, each of which can have a numeric suffix:

`alpha (aka a), beta (aka b), milestone (aka m), rc (aka cr), snapshot, "" (aka ga, final, release), sp`

All other qualifiers can have a numeric suffix, but are ordered after all known qualifiers:

e.g. `a1` <  `a2` < `beta1` < `m1` < `rc` < `ga` < `sp1` < `ANOTHERPREFIX` <  `SNAP0`

The net effect of this for Scalatest is that (using the 3.2.0 release series as an example):

`3.2.0-M1` < `3.2.0-M2` <  `3.2.0` < `3.2.0.SNAP1` < `3.2.0.SNAP2`

i.e. all applications using version ranges to resolve or compare versions will detect the `SNAP` versions as later. This will impact the Maven Versions plugin, Dependabot, Renovate etc.

Furthermore, many of those tools will treat `3.2.0-SNAP1` as a _"release"_ - i.e. it's got a version precedence higher than `ga/final`, and will propose an upgrade from a stable release version to an unstable snapshot version.

Changing to using `-SNAPSHOT1` or `ALPHA1`/`BETA1` depending on your intent will avoid this issue.

**Comment by bvenners** [comment]
Will numbering snapshots like SNAPSHOT1, SNAPSHOT2 work for sure? We originally used SNAPSHOT, which was the Maven way, and then updated the SNAPSHOT each time we did a new SNAPSHOT. We had requests back then to make those stable. If we made a SNAPSHOT1, SNAPSHOT2, and made sure we didn't release one twice, that would still address the stability goal as well as address the needs of tools like dendabot or renovate.

One other feature of SNAPn approach was that these made their way out to Maven Central. That was convenient for some, but inconvenient for others, so I think probably it is fine if they stay off of Maven Central so long as they are stable.

**Comment by timw** [comment]
TBH I haven't seen much usage of `SNAPSHOT1` etc., and with the special handling of snapshots I'm not 100% sure what it'd do.
The docs and code seem to indicate that only `-SNAPSHOT` triggers the special handling on publish and resolution (i.e. there is only one 'unstable' descriptor for a particular version) and `-SNAPSHOT1` etc. are ordered as for all other 'release' versions. I think it might be confusing for the Maven community though, given that `-SNAPSHOT` is an unstable version and `-SNAPSHOT1` would technically be a release version.

If you're trying to achieve stable pre-release coordinates, the `-alpha1`, -`beta1`, `milestone1` and `rc1` qualifiers seem to have well defined (and Maven iodomatic) behaviour.

**Comment by bvenners** [comment]
OK I'll look up what alpha1 means to people. Perhaps that is a better way to number these. We do use Mn for milestones and RCn for release candidates. the SNAPn's are really snapshots, but just intended to be stable. So maybe alpha is a fine name for those.

**Comment by daniel-shuy** [comment]
I agree with @timw. While `-SNAPSHOT1` will probably work, technically `SNAPSHOT` versions are not supposed to be released/published, so in this case `ALPHA` seems more appropriate.

**Comment by lhandrew11** [comment]
This is still causing issues for me with Dependabot unable to exclude the -SNAP[d+] versions using Gradle's versioning declaration syntax. Is there any indication on which version the SNAPs will be deprecated (if any)? Thanks :)
